.test-infra/jenkins/common_job_properties.groovy

```diff
@@ -165,21 +165,31 @@ class common_job_properties {
     }
   }
 
-  static String[] gradle_switches = [
-    // Gradle log verbosity enough to diagnose basic build issues
-    "--info",
-    // Continue the build even if there is a failure to show as many potential failures as possible.
-    '--continue',
-    // Limit background number of workers to prevent exhausting machine memory.
-    // Jenkins machines have 15GB memory, and run 2 jobs in parallel; workers are configured with
-    // JVM max heap size 3.5GB. So 2 jobs * 2 workers * 3.5GB heap = 14GB
-    '--max-workers=2',
-  ]
-
-  static void setGradleSwitches(context) {
-    for (String gradle_switch : gradle_switches) {
+  static void setGradleSwitches(context, maxWorkers = Runtime.getRuntime().availableProcessors()) {
```

Comment:
Preferably use a map like we did with Gradle applyJavaNature/... instead of parameters with default args as you can't easily add/remove default args or know what parameter your passing in.